tfhe/src/conformance.rs

```diff
@@ -0,0 +1,70 @@
+/// A trait for objects which can be checked to be conformant with a parameter set
+pub trait ParameterSetConformant {
+    type ParameterSet;
+
+    fn is_conformant(&self, parameter_set: &Self::ParameterSet) -> bool;
+}
+
+/// A constraint on a list size
+/// The list must be composed of a number `n` of groups of size `group_size` which means list size
+/// must be a multiple of `group_size`.
+/// Moreover, `n` must be:
+/// - bigger or equal to `min_inclusive_group_count`
+/// - smaller of equal to `max_inclusive_group_count`
+#[derive(Copy, Clone)]
+pub struct ListSizeConstraint {
+    min_inclusive_group_count: usize,
+    max_inclusive_group_count: usize,
+    group_size: usize,
+}
+
+impl ListSizeConstraint {
+    pub fn exact_size(size: usize) -> ListSizeConstraint {
```

Comment:
let's plan an issue to add docstrings to those functions, but otherwise I think it's good for me

Reply:
https://github.com/zama-ai/tfhe-rs-internal/issues/272